### PR TITLE
Add http protocol for Gitlab custom host

### PR DESCRIPTION
**Pull request by MartinDelille**

### CODE REVIEW

File: git-open [hunk 1]
```diff
@@ -93,14 +93,18 @@ else
...
   if [ -n "$gitlab_domain" ]; then
     if grep -q "$gitlab_domain" <<<$giturl; then
 
       # Handle GitLab's default SSH notation (like git@gitlab.domain.com:user/repo)
-      giturl=${giturl/git\@${gitlab_domain}\:/https://${gitlab_domain}/}
+      giturl=${giturl/git\@${gitlab_domain}\:/${gitlab_protocol}://${gitlab_domain}/}
 
       # handle SSH protocol (links like ssh://git@gitlab.domain.com/user/repo)
-      giturl=${giturl/#ssh\:\/\//https://}
+      giturl=${giturl/#ssh\:\/\//${gitlab_protocol}://}
```
Comment (MartinDelille) [inline git-open]: I didn't test this since my remote url use the following scheme: `git@gitlab.mydomain.com:mygroup/myrepository.git`

**Comment by ffes** [comment]
@paulirish I would suggest to merge #56 first, then look at this one. They do conflict with each other.

**Comment by MartinDelille** [comment]
@ffes As you wish!

**Comment by paulirish** [comment]
seems bad to have a source control server not on a secure connection....

File: git-open [hunk 2]
```diff
@@ -93,14 +93,18 @@ else
   # custom GitLab
   gitlab_domain=$(git config --get gitopen.gitlab.domain)
   gitlab_port=$(git config --get gitopen.gitlab.port)
+  gitlab_protocol=$(git config --get gitopen.gitlab.protocol)
+  if [ -z "$gitlab_protocol" ]; then
+      gitlab_protocol=http
```
Comment (paulirish) [inline git-open]: the default should be https, no? otherwise the docs in the readme don't make sense as setting it is redudant..

**Comment by paulirish** [comment]
@MartinDelille ive got good news and bad news. :)

good news: i'm nearly ready to merge this PR and finally get this sorted.
bad news: I edited your https branch (on your repo) which you've pointed people to.

I'm going to attempt to revert those changes to put your branch back to normal. then I'll followup with a PR to your https branch. Once accepted we'll merge that into master here. sound good?

**Comment by paulirish** [comment]
@MartinDelille k. your `http` branch is back as you left it. :)

**Comment by paulirish** [comment]
And PR to finalize all this is up: https://github.com/MartinDelille/git-open/pull/1

**Comment by MartinDelille** [comment]
Done

**Comment by paulirish** [comment]
All set! Thanks for everyone's patterns on this!

**Comment by paulirish** [comment]
Patience*

**Comment by ffes** [comment]
@paulirish we do run our gitlab on our LAN, but with https so could not test really test this. I will test the current master Monday, because I use git-open every day 😉

**Comment by paulirish** [comment]
@ffes thanks! sounds good.

I've just published 1.3.0 to npm which has this PR.

**Pull request merged**